app/controllers/posts_controller.rb

```diff
@@ -1,5 +1,30 @@
 class PostsController < ApplicationController
-  def index; end
+  def index
+    @user = User.find(params[:author_id])
+    @posts = @user.recent_posts
+  end
 
-  def show; end
+  def show
+    @user = User.find(params[:id])
+    @post = @user.posts.find(params[:id])
+    @comments = @post.comments.all
+  end
```

Comment:
You still haven't fixed the N + 1 problem.

While fetching the posts for a single user in the post#index i.e `@posts = @user.recent_posts`, you would have to query the database again to find the comments or likes for that same user on the same post and thus querying the database N additional times.

Solve this by using the `includes` method. Read more about it and see how best to implement this feature.

> HINT: It will look similar to this
`@user.posts.includes(:comments, :likes)`

Reply:
I made a mistake not committing the changes before pushing 🙏🏽